command/agent/command.go

```diff
@@ -1405,6 +1420,14 @@ General Options (clients and servers):
     interface rather than localhost for using Consul Connect. This
     mode is supported only on Linux as root.
 
+  -dev-consul
+    Starts the agent in development mode with a default Consul configuration
+    for Nomad workload identity.
```

Comment:
It might be a good idea to note that this can be combined with `-dev-connect` and doesn't automatically activate `-dev-connect`.

Reply:
Also updated `-dev-connect` to reference `-dev-consul`. I wondered if `-dev-connect` should imply `-dev-consul`, but maybe something to consider later.

Reply:
Once we completely get rid of the legacy workflow, having `-dev-connect` imply `-dev-consul` would make sense. But before that we'd be sticking people trying out legacy workflow in a bad spot (potentially including Nomad developers :grinning: )